docs/developer/best_practices/commit.rst

```diff
@@ -0,0 +1,372 @@
+
+- a topic, usually the name of the affected component (``ui``, ``build``,
+  ``docs``, …)
+
+- a slash and then the name of the sub-component (optional)
+
+- a colon
+
+- the description of the change
+
+Examples:
+
+- ``ci: use proxy-cache to reduce flakiness``
+
+- ``build/package: factorize task_dep in DEBPackage``
+
+- ``ui/volume: add banner when failed to create volume``
+
+If several components are affected:
+
+- split your commit (preferred)
+
+- pick only the most affected one
+
+- entirely omit the component (happen for truly global change, like renaming
+  ``licence`` to ``license`` over the whole codebase)
+
+As for "what is the topic?", the following heuristic works quite well for
+MetalK8s: take the name of the top-level directory (``ui``, ``salt``, ``docs``,
+…) except for ``eve`` (use ``ci`` instead). ``buildchain`` could also be
+shortened to ``build``.
+
+Having the topic in the summary line allows for faster peering over ``git log``
+output (you can know what the commit is about just by reading a few characters,
+not need to check the entire commit message or the associated diff).
+It also helps the review process: if you have a big pull request affecting
+front-end and back-end, front-end people can only review commits starting with
+``ui`` (not need to read over the whole diff, or to open each commit one by one
+in Github to see which ones are interesting).
+
+Body
+::::
+
+The body should answer the following questions:
+
+- Why did you make this change? (is this for a new feature, a bugfix — then,
+  why was it buggy? —, some cleanup, some optimization, …).
+  It is really important to describe the intent/motivation behind the changes.
+
+- What change did you make? Document what the original problem was and how it
+  is being fixed (can be omitted for short obvious patches).
+
+- Why did you make the change in that way and not in another (mention alternate
+  solutions considered but discarded, if any)?
+
+When writing your message you must consider that your reader does not know
+anything about the code you have patched.
+
+You should also describe any limitations of the current code. This will avoid
+reviewer pointing them out, and also inform future people looking at the code
+which tradeoffs were made at the time.
+
+Lines should be wrapped at 72 characters.
+
+Footer
+::::::
+
+Use references such as ``Refs``, ``See``, ``Fixes`` or ``Closes`` followed by
+an issue number to automate issue management.
+
+In addition to the references, you can also provide the URLs (it will be
+quicker to access them from the terminal).
+
+Example:
+
+.. code:: text
+
+    topic: description
+
+    [… commit message body …]
+
+    Refs: #XXXXX
+    Refs: #YYYYY
+    Closes: #ZZZZZ
+    See: https://github.com/scality/metalk8s/issues/XXXXX
+    See: https://github.com/scality/metalk8s/issues/YYYYY
+    See: https://github.com/scality/metalk8s/issues/ZZZZZ
+
+Footer can also contains signature (``git commit -s``) or cherry-pick source
```

Comment:
Footer can also contain a signature??

Reply:
Yeah they could.
Most of my commits end with `Signed-off-by: Sylvain Laperche <…> `

And actually, it's even required for contributors outside of Scality: https://github.com/scality/metalk8s/blame/development/2.4/CONTRIBUTING.md#L110-L121

Reply:
Yes yes...
Here, it was more of an issue with
`s/contains/contain a/`

Reply:
Oh right. Done.